test/k8sT/Updates.go

```diff
@@ -207,8 +210,38 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
 			helpers.KubeSystemNamespace, l7Policy, helpers.KubectlApply, timeout)
 		Expect(err).Should(BeNil(), "cannot import l7 policy: %v", l7Policy)
 
+		_, err = kubectl.CiliumPolicyAction(
+			helpers.KubeSystemNamespace, l4Policy, helpers.KubectlApply, timeout)
+		Expect(err).Should(BeNil(), "cannot import l7 policy: %v", l7Policy)
+
 		validateEndpointsConnection()
 
+		By("Starting background process that checks L4 connectivity each 5 seconds")
+
+		appPods := helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+		podsIps, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, "zgroup=testapp")
+		Expect(err).To(BeNil(), "cannot get pods ips for testapp")
+		app4IP := podsIps[appPods[helpers.App4]]
+
+		ctx, cancel := context.WithCancel(context.Background())
+		restartConnections := []*helpers.CmdRes{}
+		go func() {
+			ticker := time.NewTicker(time.Duration(5 * time.Second))
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					res := kubectl.ExecPodCmd(
+						helpers.DefaultNamespace,
+						appPods[helpers.App2],
+						helpers.CurlFail("http://%s/public", app4IP))
```

Comment:
1. Do a curl do the podIP as well as to the service name
2. Also test DNS lookup manually
3. Test connectivity to the k8s apiserver with a CIDR policy allowing the connection to the apiserver

Reply:
Added DNS lookup + using clusterIP and egress rule. 

In case of failed request DNS, all the output in background will be saved, so it'll be easy to debug.